frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -897,7 +897,7 @@ function VirtualKeyboard:setLayer(key)
         self.umlautmode = not self.umlautmode
     end
     self:initLayer()
-    self:_refresh(true)
+    self:_refresh(key ~= "Shift")
```

Comment:
> I mean darkening of all the keyboard (similar to flashing) when switching to the sym layout.
When switching to the shift layout there is no flashing, just the labels change.

I guess it might be because of this. Which is mentionned in the very first post above:

> The only way to make the whole keyboard not flash when switching layers was to turn on "Avoid mandatory black flashes in UI" which messes up a lot of other things, so I made it only flash when going to symbol mode and not flash when shifting.

Not understanding much about the issue/reasons. @MoreFoxBeans @NiLuJe ?

Reply:
> I guess it might be because of this.

Thanks, exactly. I've changed that to `self:_refresh(false)`, and it flashes only when switching language with geometry change, much better.

Reply:
> Thanks, exactly. I've changed that to `self:_refresh(false)`, and it flashes only when switching language with geometry change, much better.

Interesting, I don't see any flashes at all on Kindle and it looks quite OK even after multiple switches. I'll add this change in current keyboard version.

Reply:
I kinda had this set to `true` on purpose, so I'm not necessarily a fan of any other approaches ;p.

Then again, I barely use the keyboard, so it's not a hill I'm willing to die on, and true or false is much better than a weird true-but-maybe-not ;).

And, IIRC, the main reason behind making this true (key labels being lost in refresh storms) has gotten slightly better than when I did this ;).

Reply:
> I kinda had this set to `true` on purpose, so I'm not necessarily a fan of any other approaches ;p.

It seems to me that `true` or `false` have no meaningful difference right now. Isn't it broken? 
With either of these, keyboard does not completely flash on my Onyx Boox Kon-Tiki 2 (and I don't remember it did on Kindle PW3 too, but I am not sure). I can only see the key borders to be drawn to solid black then to the usual gray when I switch layouts.
`Flash keyboard` is set to `on`, if that matters.

Reply:
Let's set it to `false` for a while, until any issue is reported.